openedx/features/course_experience/templates/course_experience/course-outline-fragment.html

```diff
@@ -35,9 +35,32 @@ <h3 class="section-title">${ section['display_name'] }</h3>
                                 >
                                     <div class="subsection-text">
                                         ## Subsection title
-                                        <span class="subsection-title">${ subsection['display_name'] }</span>
-
+                                        <span class="subsection-title">
+                                            % if subsection['id'] in milestones:
+                                                % if milestones[subsection['id']]['completed_prereqs']:
+                                                    <span class="menu-icon icon fa fa-unlock"
+                                                        aria-hidden="true">
+                                                    </span>
+                                                % else:
+                                                    <span class="menu-icon icon fa fa-lock" 
+                                                    aria-hidden="true">
+                                                    </span>
+                                                % endif
+                                            % endif
+                                            <span class="subsection-title-name">
+                                                ${ subsection['display_name'] }
+                                            </span>
+                                            % if subsection['id'] in milestones: 
+                                                % if milestones[subsection['id']]['completed_prereqs']:
+                                                    <span class="sr">&nbsp;${_("Unlocked")}</span>
+                                                % else:
+                                                    <span class="details">
+                                                        ${ _("(Prerequisite required)") }
+                                                    </span>
+                                                % endif 
+                                            % endif
```

Comment:
Nit: Sometimes we take DRY too far forsaking readability.  For example, which of the following is more readable?
```
if A:
  1
else:
  2
3
if A:
  4
else:
  5
```

versus

```
if A:
  1
  3
  4
else:
  2
  3
  5
```

Reply:
I was trying to avoid duplicating the middle thing (3 in your example), but this is a good point - your second example is more readable.